### PR TITLE
keystore cryptoengine

### PR DESCRIPTION
**Pull request by lahosken**
We have a few "CryptoEngines" that differ mostly in how they store keys; they're about the same in how they generate keys, make signers, make verifiers.

So introduce concept of KeyStore and make MemoryCryptoEngine, BrowserCryptoEngine, and UnsecuredFileCryptoEngine into thin wrappers around new classes MemoryKeyStore, BrowserKeyStore, and UnsecuredFileKeyStore.

This changes the API for MemoryCryptoEngine and BrowserCryptoEngine: their APIs are now async.

Changed the schema BrowserCryptoEngine uses to keep keys in localStorage.

### CODE REVIEW

File: test/security/engines/KeyStore.spec.js [hunk 1]
```diff
@@ -0,0 +1,199 @@
+const chai = require('chai');
```
Comment (lahosken) [inline test/security/engines/KeyStore.spec.js]: this was in its own dir, test/security/keystores/KeyStore.spec.js . I wasn't sure whether to mv it to test/security/engines or mv the KeyStore classes to src/security/keystores . I flipped a coin and it said mv this to test/security/engines. But I could be persuaded to not-follow the coin's advice; the coin isn't always right.

File: src/security/engines/BrowserKeyStore.js [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+            window.localStorage.schemaVersion = JSON.stringify(config.localStorageSchemaVersion);
+        }
+    }
+
+    /**
+     * Store a member's keypair.
+     *
+     * @param {string} memberId - ID of member
+     * @param {Object} keypair - keypair to store
+     * @return {Object} keypair - same ketypair
```
Comment (mariano54) [inline src/security/engines/BrowserKeyStore.js]: typo

**Comment by lahosken** [comment]
> Have you confirmed that all tests are passing, including sdk-js-tests

Pleasantly-surprisingly, sdk-js-tests passed w/no changes. (I guessed I'd need to add an await somewhere).

**Comment by lahosken** [comment]
bamboo kindly points out that Object.values is newfangled weirdness. will fix.

**Pull request merged**